### ISSUE TITLE
Drop Android 4.0.x support for app release 3.4.0

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#2923]
I'd like to slowly but steadily remove support for (very) old Android versions and would like to start with dropping 4.0.x making 4.1 the minimum version (baby steps). Looking at the numbers the app lost 16% of the total 4.0.x active installations dropping down to 161 devices, so I think we are safe to remove the support after 3.3.x which would then be the last feature release which is the first one to support Nc14 features.

Here is the actual 4.0.x active installations graph for the last month:
![android_4_0_x](https://user-images.githubusercontent.com/1315170/44570865-7c20a700-a77f-11e8-885d-e69137f4a683.png)

What do you think @mario @ardevd @tobiasKaminsky

**Comment by ardevd** [comment]
I absolutely support this! I would probably make 5.0 the minimum version but as you say - baby steps!

**Comment by mario** [comment]
5.0 pls.

**Comment by AndyScherzinger** [comment]
Well, dropping 4.x and making 5.x the minimum version would drop support for 3,4% of the active users where 4.4 has the same amount of users as 5.0 (with both having more than 5000 users each!). That is why I thought of baby steps since I don't see a good reason to drop support for >5000 people.

Easier maintenance isn't an argument from my point of view if nobody finds the time to take care of improving the auto upload which is where > Android 5.0 might be helpful. So´in my opinion just dropping 4.x and leaving the actual features as-is, is just dropping support for a lot of people for no reason.

**Comment by AndyScherzinger** [comment]
Which is why I opened the issue for **dropping_4.0.x_only**

**Comment by rullzer** [comment]
4.0 is fine and totally defendable IMO. And indeed baby steps.

**Comment by AndyScherzinger** [comment]
@oparoz can you clarify on this, especially talking about 4.0.x (since it has been release Oct. 2011, overall market share <0,3%, for Nc it's <0,1% meaning 150 devices, declining by about 20% each month).

**Comment by AndyScherzinger** [comment]
@oparoz @tobiasKaminsky any news and feedback on dropping Android 4.0.x making Android 4.1 the minimum requirement?

**Comment by tobiasKaminsky** [comment]
From my side, last info is that 4.x is required by a customer and thus we have to support it.
@oparoz, please clarify

**Comment by rullzer** [comment]
Also note that google will most likely drop support for chrome in anything below 4.4 soon.

**Comment by tobiasKaminsky** [comment]
@oparoz what is our least supported version?
4.0.x is really a bit old…
So I agree that we should remove this, if possible.

**Comment by rullzer** [comment]
Google is even dropping support for it. So play services will no longer work.

**Comment by tobiasKaminsky** [comment]
@rullzer do you have a reference for this?

**Comment by rullzer** [comment]
https://android-developers.googleblog.com/2018/12/google-play-services-discontinuing.html

**Comment by AndyScherzinger** [comment]
Again, and again and again: @tobiasKaminsky @oparoz can we get an answer on this? Keeping this blocked by not deciding on it since over a year is just sad
<img width="710" alt="2019-09-17 08_41_04-Statistics - Nextcloud - Google Play Console" src="https://user-images.githubusercontent.com/1315170/65017410-faba2a80-d926-11e9-8d90-b3b850783c05.png">

**Comment by tobiasKaminsky** [comment]
What about this idea?
If oldest version is <0.5% then we drop it? 
https://developer.android.com/about/dashboards/index.html

So we would right now drop 4.0.x and have min 16?

**Comment by tobiasKaminsky** [comment]
Right, they updated it in May and now recommend to use our own statistics available in dev console.

so this are our oldest android version with active installations:
![image](https://user-images.githubusercontent.com/5836855/67081446-72ce7700-f197-11e9-90d8-da1fd65a7daf.png)

Which threshold shall we use?
I would like to have decision that we can use to check just before when releasing a RC1.

We have right now 400k active installations. so 400 users are 0.1%.
Shall we take this as a threshold?
That means we can right now deprecate 4.0.x, but maybe on 3.9 RC1 even 4.1.

(we cannot deprecate 4.3 as it is only possible to specify a min version, nothing inbetween).

If we agree on this, we should make this info available on release.
Users with older devices still can use the old versions…

**Comment by AndyScherzinger** [comment]
So which threshold are you proposing @tobiasKaminsky? Your last 2 comments propose different one: 0.1 and 0.5.

Personally I'd say 0.1 is too little but for 3.10 I'd say we can make v16 the min version. 
Since 3.9 RC1 is in 10 days I would start dropping support yet since we want to do some housekeeping on the way. 

What do you think? 

After that I'd even go as far as to make Android 4.4 the min version for 3.11.

**Comment by rullzer** [comment]
0.1% seems very low to me ;) We don't even support browsers with that small percentage.

IMO: 0.5%  && EOL seems fair.
But I'm also fie to start with 0.1% if you feel more confortable with that. As long as we start with something.

**Comment by AndyScherzinger** [comment]
I am with @rullzer on the threshold, <=0,5% and EOL

**Comment by tobiasKaminsky** [comment]
Well then let us use 0.5% for the version after this (3.10).

I update the overview issue then.
